// @aws-sdk/client-organizations/dist-es/commands/ListHandshakesForOrganizationCommand.js
import { getEndpointPlugin } from "@smithy/middleware-endpoint";
import { getSerdePlugin } from "@smithy/middleware-serde";
import { Command as $Command } from "@smithy/smithy-client";
import { SMITHY_CONTEXT_KEY, } from "@smithy/types";
import { ListHandshakesForOrganizationResponseFilterSensitiveLog, } from "../models/models_0";
import { de_ListHandshakesForOrganizationCommand, se_ListHandshakesForOrganizationCommand, } from "../protocols/Aws_json1_1";
export { $Command };
export class ListHandshakesForOrganizationCommand extends $Command {
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, ListHandshakesForOrganizationCommand.getEndpointParameterInstructions()));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "OrganizationsClient";
        const commandName = "ListHandshakesForOrganizationCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: (_) => _,
            outputFilterSensitiveLog: ListHandshakesForOrganizationResponseFilterSensitiveLog,
            [SMITHY_CONTEXT_KEY]: {
                service: "AWSOrganizationsV20161128",
                operation: "ListHandshakesForOrganization",
            },
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return se_ListHandshakesForOrganizationCommand(input, context);
    }
    deserialize(output, context) {
        return de_ListHandshakesForOrganizationCommand(output, context);
    }
}
