// @aws-sdk/client-organizations/dist-cjs/Organizations.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Organizations = void 0;
const smithy_client_1 = require("@smithy/smithy-client");
const AcceptHandshakeCommand_1 = require("./commands/AcceptHandshakeCommand");
const AttachPolicyCommand_1 = require("./commands/AttachPolicyCommand");
const CancelHandshakeCommand_1 = require("./commands/CancelHandshakeCommand");
const CloseAccountCommand_1 = require("./commands/CloseAccountCommand");
const CreateAccountCommand_1 = require("./commands/CreateAccountCommand");
const CreateGovCloudAccountCommand_1 = require("./commands/CreateGovCloudAccountCommand");
const CreateOrganizationalUnitCommand_1 = require("./commands/CreateOrganizationalUnitCommand");
const CreateOrganizationCommand_1 = require("./commands/CreateOrganizationCommand");
const CreatePolicyCommand_1 = require("./commands/CreatePolicyCommand");
const DeclineHandshakeCommand_1 = require("./commands/DeclineHandshakeCommand");
const DeleteOrganizationalUnitCommand_1 = require("./commands/DeleteOrganizationalUnitCommand");
const DeleteOrganizationCommand_1 = require("./commands/DeleteOrganizationCommand");
const DeletePolicyCommand_1 = require("./commands/DeletePolicyCommand");
const DeleteResourcePolicyCommand_1 = require("./commands/DeleteResourcePolicyCommand");
const DeregisterDelegatedAdministratorCommand_1 = require("./commands/DeregisterDelegatedAdministratorCommand");
const DescribeAccountCommand_1 = require("./commands/DescribeAccountCommand");
const DescribeCreateAccountStatusCommand_1 = require("./commands/DescribeCreateAccountStatusCommand");
const DescribeEffectivePolicyCommand_1 = require("./commands/DescribeEffectivePolicyCommand");
const DescribeHandshakeCommand_1 = require("./commands/DescribeHandshakeCommand");
const DescribeOrganizationalUnitCommand_1 = require("./commands/DescribeOrganizationalUnitCommand");
const DescribeOrganizationCommand_1 = require("./commands/DescribeOrganizationCommand");
const DescribePolicyCommand_1 = require("./commands/DescribePolicyCommand");
const DescribeResourcePolicyCommand_1 = require("./commands/DescribeResourcePolicyCommand");
const DetachPolicyCommand_1 = require("./commands/DetachPolicyCommand");
const DisableAWSServiceAccessCommand_1 = require("./commands/DisableAWSServiceAccessCommand");
const DisablePolicyTypeCommand_1 = require("./commands/DisablePolicyTypeCommand");
const EnableAllFeaturesCommand_1 = require("./commands/EnableAllFeaturesCommand");
const EnableAWSServiceAccessCommand_1 = require("./commands/EnableAWSServiceAccessCommand");
const EnablePolicyTypeCommand_1 = require("./commands/EnablePolicyTypeCommand");
const InviteAccountToOrganizationCommand_1 = require("./commands/InviteAccountToOrganizationCommand");
const LeaveOrganizationCommand_1 = require("./commands/LeaveOrganizationCommand");
const ListAccountsCommand_1 = require("./commands/ListAccountsCommand");
const ListAccountsForParentCommand_1 = require("./commands/ListAccountsForParentCommand");
const ListAWSServiceAccessForOrganizationCommand_1 = require("./commands/ListAWSServiceAccessForOrganizationCommand");
const ListChildrenCommand_1 = require("./commands/ListChildrenCommand");
const ListCreateAccountStatusCommand_1 = require("./commands/ListCreateAccountStatusCommand");
const ListDelegatedAdministratorsCommand_1 = require("./commands/ListDelegatedAdministratorsCommand");
const ListDelegatedServicesForAccountCommand_1 = require("./commands/ListDelegatedServicesForAccountCommand");
const ListHandshakesForAccountCommand_1 = require("./commands/ListHandshakesForAccountCommand");
const ListHandshakesForOrganizationCommand_1 = require("./commands/ListHandshakesForOrganizationCommand");
const ListOrganizationalUnitsForParentCommand_1 = require("./commands/ListOrganizationalUnitsForParentCommand");
const ListParentsCommand_1 = require("./commands/ListParentsCommand");
const ListPoliciesCommand_1 = require("./commands/ListPoliciesCommand");
const ListPoliciesForTargetCommand_1 = require("./commands/ListPoliciesForTargetCommand");
const ListRootsCommand_1 = require("./commands/ListRootsCommand");
const ListTagsForResourceCommand_1 = require("./commands/ListTagsForResourceCommand");
const ListTargetsForPolicyCommand_1 = require("./commands/ListTargetsForPolicyCommand");
const MoveAccountCommand_1 = require("./commands/MoveAccountCommand");
const PutResourcePolicyCommand_1 = require("./commands/PutResourcePolicyCommand");
const RegisterDelegatedAdministratorCommand_1 = require("./commands/RegisterDelegatedAdministratorCommand");
const RemoveAccountFromOrganizationCommand_1 = require("./commands/RemoveAccountFromOrganizationCommand");
const TagResourceCommand_1 = require("./commands/TagResourceCommand");
const UntagResourceCommand_1 = require("./commands/UntagResourceCommand");
const UpdateOrganizationalUnitCommand_1 = require("./commands/UpdateOrganizationalUnitCommand");
const UpdatePolicyCommand_1 = require("./commands/UpdatePolicyCommand");
const OrganizationsClient_1 = require("./OrganizationsClient");
const commands = {
    AcceptHandshakeCommand: AcceptHandshakeCommand_1.AcceptHandshakeCommand,
    AttachPolicyCommand: AttachPolicyCommand_1.AttachPolicyCommand,
    CancelHandshakeCommand: CancelHandshakeCommand_1.CancelHandshakeCommand,
    CloseAccountCommand: CloseAccountCommand_1.CloseAccountCommand,
    CreateAccountCommand: CreateAccountCommand_1.CreateAccountCommand,
    CreateGovCloudAccountCommand: CreateGovCloudAccountCommand_1.CreateGovCloudAccountCommand,
    CreateOrganizationCommand: CreateOrganizationCommand_1.CreateOrganizationCommand,
    CreateOrganizationalUnitCommand: CreateOrganizationalUnitCommand_1.CreateOrganizationalUnitCommand,
    CreatePolicyCommand: CreatePolicyCommand_1.CreatePolicyCommand,
    DeclineHandshakeCommand: DeclineHandshakeCommand_1.DeclineHandshakeCommand,
    DeleteOrganizationCommand: DeleteOrganizationCommand_1.DeleteOrganizationCommand,
    DeleteOrganizationalUnitCommand: DeleteOrganizationalUnitCommand_1.DeleteOrganizationalUnitCommand,
    DeletePolicyCommand: DeletePolicyCommand_1.DeletePolicyCommand,
    DeleteResourcePolicyCommand: DeleteResourcePolicyCommand_1.DeleteResourcePolicyCommand,
    DeregisterDelegatedAdministratorCommand: DeregisterDelegatedAdministratorCommand_1.DeregisterDelegatedAdministratorCommand,
    DescribeAccountCommand: DescribeAccountCommand_1.DescribeAccountCommand,
    DescribeCreateAccountStatusCommand: DescribeCreateAccountStatusCommand_1.DescribeCreateAccountStatusCommand,
    DescribeEffectivePolicyCommand: DescribeEffectivePolicyCommand_1.DescribeEffectivePolicyCommand,
    DescribeHandshakeCommand: DescribeHandshakeCommand_1.DescribeHandshakeCommand,
    DescribeOrganizationCommand: DescribeOrganizationCommand_1.DescribeOrganizationCommand,
    DescribeOrganizationalUnitCommand: DescribeOrganizationalUnitCommand_1.DescribeOrganizationalUnitCommand,
    DescribePolicyCommand: DescribePolicyCommand_1.DescribePolicyCommand,
    DescribeResourcePolicyCommand: DescribeResourcePolicyCommand_1.DescribeResourcePolicyCommand,
    DetachPolicyCommand: DetachPolicyCommand_1.DetachPolicyCommand,
    DisableAWSServiceAccessCommand: DisableAWSServiceAccessCommand_1.DisableAWSServiceAccessCommand,
    DisablePolicyTypeCommand: DisablePolicyTypeCommand_1.DisablePolicyTypeCommand,
    EnableAllFeaturesCommand: EnableAllFeaturesCommand_1.EnableAllFeaturesCommand,
    EnableAWSServiceAccessCommand: EnableAWSServiceAccessCommand_1.EnableAWSServiceAccessCommand,
    EnablePolicyTypeCommand: EnablePolicyTypeCommand_1.EnablePolicyTypeCommand,
    InviteAccountToOrganizationCommand: InviteAccountToOrganizationCommand_1.InviteAccountToOrganizationCommand,
    LeaveOrganizationCommand: LeaveOrganizationCommand_1.LeaveOrganizationCommand,
    ListAccountsCommand: ListAccountsCommand_1.ListAccountsCommand,
    ListAccountsForParentCommand: ListAccountsForParentCommand_1.ListAccountsForParentCommand,
    ListAWSServiceAccessForOrganizationCommand: ListAWSServiceAccessForOrganizationCommand_1.ListAWSServiceAccessForOrganizationCommand,
    ListChildrenCommand: ListChildrenCommand_1.ListChildrenCommand,
    ListCreateAccountStatusCommand: ListCreateAccountStatusCommand_1.ListCreateAccountStatusCommand,
    ListDelegatedAdministratorsCommand: ListDelegatedAdministratorsCommand_1.ListDelegatedAdministratorsCommand,
    ListDelegatedServicesForAccountCommand: ListDelegatedServicesForAccountCommand_1.ListDelegatedServicesForAccountCommand,
    ListHandshakesForAccountCommand: ListHandshakesForAccountCommand_1.ListHandshakesForAccountCommand,
    ListHandshakesForOrganizationCommand: ListHandshakesForOrganizationCommand_1.ListHandshakesForOrganizationCommand,
    ListOrganizationalUnitsForParentCommand: ListOrganizationalUnitsForParentCommand_1.ListOrganizationalUnitsForParentCommand,
    ListParentsCommand: ListParentsCommand_1.ListParentsCommand,
    ListPoliciesCommand: ListPoliciesCommand_1.ListPoliciesCommand,
    ListPoliciesForTargetCommand: ListPoliciesForTargetCommand_1.ListPoliciesForTargetCommand,
    ListRootsCommand: ListRootsCommand_1.ListRootsCommand,
    ListTagsForResourceCommand: ListTagsForResourceCommand_1.ListTagsForResourceCommand,
    ListTargetsForPolicyCommand: ListTargetsForPolicyCommand_1.ListTargetsForPolicyCommand,
    MoveAccountCommand: MoveAccountCommand_1.MoveAccountCommand,
    PutResourcePolicyCommand: PutResourcePolicyCommand_1.PutResourcePolicyCommand,
    RegisterDelegatedAdministratorCommand: RegisterDelegatedAdministratorCommand_1.RegisterDelegatedAdministratorCommand,
    RemoveAccountFromOrganizationCommand: RemoveAccountFromOrganizationCommand_1.RemoveAccountFromOrganizationCommand,
    TagResourceCommand: TagResourceCommand_1.TagResourceCommand,
    UntagResourceCommand: UntagResourceCommand_1.UntagResourceCommand,
    UpdateOrganizationalUnitCommand: UpdateOrganizationalUnitCommand_1.UpdateOrganizationalUnitCommand,
    UpdatePolicyCommand: UpdatePolicyCommand_1.UpdatePolicyCommand,
};
class Organizations extends OrganizationsClient_1.OrganizationsClient {
}
exports.Organizations = Organizations;
(0, smithy_client_1.createAggregatedClient)(commands, Organizations);
